flink-yarn/src/main/java/org/apache/flink/yarn/YarnApplicationFileUploader.java

```diff
@@ -438,6 +446,18 @@ private static boolean isPlugin(Path path) {
         return false;
     }
 
+    private static boolean isUsrLib(Path path) {
```

Comment:
I suggest to introduce the `isUsrLibDirIncludedInProvidedLib` instead and then `checkArgument` in the constructor of `YarnApplicationFileUploader`. Because the files in the provided lib will be relativized, so we just need to check the name of provided lib.

```
    private boolean isUsrLibDirIncludedInProvidedLib(final List<Path> providedLibDirs)
            throws IOException {
        for (Path path : providedLibDirs) {
            final FileStatus fileStatus = fileSystem.getFileStatus(path);
            if (fileStatus.isDirectory()
                    && ConfigConstants.DEFAULT_FLINK_USR_LIB_DIR.equals(path.getName())) {
                return true;
            }
        }
        return false;
    }
```

Reply:
Make sense. Fixed in ef3f0104fc31def407b74b4698d5f7dd0404a333